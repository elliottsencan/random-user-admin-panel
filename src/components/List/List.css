.list {
	display:flex;
	flex-direction:column;
	flex: 1;
	background-color: #2C3241;
	padding: 0 2em;
	border-radius: 30px;
}

.list header{
	flex: 0 0 4em;
	align-items: center;
	text-align: left;
	display: flex;
	padding-top: .75em;
	margin: 0 1.5em;
	border-bottom: 1px solid rgba(255,255,255,0.4);
}

.list header h4 {
	margin: 0;
	font-size: 20px;
	flex: 4;
}

.list header > div {
	flex: 1;
}

.list table {
	margin: 0 1em;
	padding: 1.5em 0;
}

.list table tr{
	color: #535A6B;
}

.list table tr.active{
	color: #ffffff;
}

.list table td {
	padding: 0.25em;
  font-size: 18px;
}

.list table td.button-data-cell {
	text-align: right;
	vertical-align: top;
}

.list table td.subtext-data-cell {
	padding-bottom: 1.5em;
	padding-top: 0;
	font-size: 14px;
}

.list table td .button-subtext-cell {
	font-size: 12px;
	padding-top:2px;
}