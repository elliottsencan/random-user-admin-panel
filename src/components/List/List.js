import React, { Fragment, Component } from 'react';
import Button from '../Button/Button';
import Search from '../Search/Search';
import Popover from '../Popover/Popover';
import './List.css';

//render prop candidate
export default class List extends Component {
  render() {
    return ( <section className="list">
      <header>
        <h4>{this.props.title}</h4>
        <div className="search-bar">
          <Search/>
        </div>
        <div className="filter-control">
          <Popover
            items={[
              {
                text: 'Number'
              }, {
                text: 'State'
              }
            ]}
            label="Filter By"/></div>
      </header>
      <table>
        <tbody>
          {
            this.props.data.map( item => {
              return ( <Fragment>
                <tr className={item.active
                    ? 'active'
                    : ''}>
                  <td>{item.text}</td>
                  <td rowSpan="2" className="button-data-cell">
                    <Button {...item.button}/> {item.buttonSubtext && <div className="button-subtext-cell">{item.buttonSubtext}</div>}
                  </td>
                </tr>
                <tr className={item.active
                    ? 'active'
                    : ''}>
                  <td className="subtext-data-cell">{item.subtext}</td>
                </tr>
              </Fragment> );
            } )
          }
        </tbody>
      </table>
    </section> );
  }
}