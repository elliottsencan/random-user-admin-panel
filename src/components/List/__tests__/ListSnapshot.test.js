import React from 'react';
import List from '../List';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure( { adapter: new Adapter() } );

it( 'renders correctly', () => {
  const tree = toJson( shallow( <List/> ) );
  expect( tree ).toMatchSnapshot();
} );