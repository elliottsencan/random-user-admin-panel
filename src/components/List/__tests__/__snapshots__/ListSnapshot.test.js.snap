// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="list"
>
  <header>
    <h4 />
    <div
      className="search-bar"
    >
      <Component
        iconStyle={
          Object {
            "color": "#8D93A4",
          }
        }
      />
    </div>
    <div
      className="filter-control"
    >
      <Component
        buttonProps={
          Object {
            "backgroundColor": "#3A4051",
            "icon": <CommunicationClearAll
              color="#8D93A4"
          />,
            "labelPosition": "after",
            "labelStyle": Object {
              "color": "#8D93A4",
              "textTransform": "capitalize",
            },
            "style": Object {
              "borderRadius": "20px",
            },
          }
        }
        items={Array []}
        label="Filter By"
        onChangeHandler={[Function]}
      />
    </div>
  </header>
  <table>
    <tbody />
  </table>
</section>
`;
